### PR TITLE
MF-1347 - Add policies

### LINKED ISSUE #1347: Add policies
**FEATURE REQUEST**

1. Is there an open issue addressing this request? If it does, please add a "+1" reaction to the
   existing issue, otherwise proceed to step 2.

2. Describe the feature you are requesting, as well as the possible use case(s) for it.
Following the https://github.com/mainflux/mainflux/issues/931 and https://github.com/mainflux/mainflux/issues/1346, we need to add policies as super-important feature.

3. Indicate the importance of this feature to you (must-have, should-have, nice-to-have).
Must-have

### PR DESCRIPTION
**Pull request by mteodor**
### What does this do?
Introducing access policies 

### Which issue(s) does this PR fix/relate to?
Resolves #1347

### CODE REVIEW

File: vendor/modules.txt [hunk 1]
```diff
@@ -9,8 +9,16 @@ github.com/Microsoft/go-winio
 github.com/Microsoft/go-winio/pkg/guid
 # github.com/Nvveen/Gotty v0.0.0-20120604004816-cd527374f1e5
 github.com/Nvveen/Gotty
+# github.com/OneOfOne/xxhash v1.2.8
+github.com/OneOfOne/xxhash
```
Comment (drasko) [inline vendor/modules.txt]: Where do we use this? Is this OPA dependency?

File: vendor/modules.txt [hunk 2]
```diff
@@ -9,8 +9,16 @@ github.com/Microsoft/go-winio
...
 # github.com/beorn7/perks v1.0.1
 github.com/beorn7/perks/quantile
+# github.com/bytecodealliance/wasmtime-go v0.24.0
```
Comment (drasko) [inline vendor/modules.txt]: Same here.

File: vendor/modules.txt [hunk 3]
```diff
@@ -46,6 +54,8 @@ github.com/fatih/color
 github.com/fsnotify/fsnotify
 # github.com/fxamacker/cbor/v2 v2.2.0
 github.com/fxamacker/cbor/v2
+# github.com/ghodss/yaml v1.0.0
+github.com/ghodss/yaml
```
Comment (drasko) [inline vendor/modules.txt]: And here

File: vendor/modules.txt [hunk 4]
```diff
@@ -73,6 +83,15 @@ github.com/go-stack/stack
 # github.com/go-zoo/bone v1.3.0
 ## explicit
 github.com/go-zoo/bone
+# github.com/gobwas/glob v0.2.3
```
Comment (drasko) [inline vendor/modules.txt]: And here

File: vendor/modules.txt [hunk 5]
```diff
@@ -372,6 +450,12 @@ github.com/x448/float16
 github.com/xdg/scram
 # github.com/xdg/stringprep v1.0.1-0.20180714160509-73f8eece6fdc
 github.com/xdg/stringprep
+# github.com/xeipuuv/gojsonpointer v0.0.0-20190905194746-02993c407bfb
+github.com/xeipuuv/gojsonpointer
```
Comment (drasko) [inline vendor/modules.txt]: And here

File: vendor/modules.txt [hunk 6]
```diff
@@ -372,6 +450,12 @@ github.com/x448/float16
...
+# github.com/xeipuuv/gojsonreference v0.0.0-20180127040603-bd5ef7bd5415
+github.com/xeipuuv/gojsonreference
+# github.com/yashtewari/glob-intersection v0.0.0-20180916065949-5c77d914dd0b
+github.com/yashtewari/glob-intersection
```
Comment (drasko) [inline vendor/modules.txt]: And here

**Review by drasko — CHANGES_REQUESTED** [review]

File: vendor/modules.txt (hunk 1 above)
Comment (mteodor) [reply vendor/modules.txt]: yes

File: vendor/modules.txt (hunk 2 above)
Comment (mteodor) [reply vendor/modules.txt]: yes

File: vendor/modules.txt (hunk 3 above)
Comment (mteodor) [reply vendor/modules.txt]: yes

File: vendor/modules.txt (hunk 4 above)
Comment (mteodor) [reply vendor/modules.txt]: yes

File: vendor/modules.txt (hunk 5 above)
Comment (mteodor) [reply vendor/modules.txt]: yes

File: vendor/modules.txt (hunk 6 above)
Comment (mteodor) [reply vendor/modules.txt]: yes

**Review by nmarcetic — CHANGES_REQUESTED** [review]
@mteodor @drasko 
This is impossible to review without spending hours just to figure out an idea. Can you please document/elaborate the idea, if you don't want to use [MIP"s](https://github.com/mainflux/mips) at least describe what you are doing how and why in the PR description?

**Comment by drasko** [comment]
@mteodor will write a detailed MIP

**Comment by nmarcetic** [comment]
> @mteodor will write a detailed MIP

Great! Thanks

**Comment by drasko** [comment]
@mteodor what about this one - should it be closed in favor of https://github.com/mainflux/mainflux/pull/1440?

**Comment by mteodor** [comment]
> @mteodor what about this one - should it be closed in favor of #1440?
Yes, we can close this one

**Comment by drasko** [comment]
Closing in favor of https://github.com/mainflux/mainflux/pull/1440

**Pull request closed without merging**